demo/components/typeahead/typeahead-demo.ts

```diff
@@ -50,32 +52,14 @@ export class TypeaheadDemo {
     {id: 49, name: 'West Virginia'}, {id: 50, name: 'Wisconsin'},
     {id: 51, name: 'Wyoming'}];
 
-  private getContext() {
-    return this;
-  }
-
-  private _cache:any;
-  private _prevContext:any;
-
-  private getAsyncData(context:any):Function {
-    if (this._prevContext === context) {
-      return this._cache;
-    }
+  constructor() {
+    this.dataSource = Observable.create((observer:any) => {
+      let query = new RegExp(this.asyncSelected, 'ig');
 
-    this._prevContext = context;
-    let f:Function = function ():Promise<string[]> {
-      let p:Promise<string[]> = new Promise((resolve:Function) => {
-        setTimeout(() => {
-          let query = new RegExp(context.asyncSelected, 'ig');
-          return resolve(context.states.filter((state:any) => {
-            return query.test(state);
-          }));
-        }, 200);
-      });
-      return p;
-    };
-    this._cache = f;
-    return this._cache;
+      observer.next(this.states.filter((state:any) => {
```

Comment:
@valorkin we put filtered options to the component in case of async mode (same as typeahead for angular 1)